src/sarif_converter.ts

```diff
@@ -187,3 +203,39 @@ const getResults = (suggestions: ISarifSuggestions) => {
   }
   return output;
 };
+
+//custom string splice implementation
+function spliceSlice(str: string, index: number, count: number, add: string) {
+  // We cannot pass negative indexes directly to the 2nd slicing operation.
+  if (index < 0) {
+    index = str.length + index;
+    if (index < 0) {
+      index = 0;
+    }
+  }
+
+  return str.slice(0, index) + (add || '') + str.slice(index + count);
+}
+
+const getArgumentsAndMessage2 = (
+  helpers: { index: number[]; msg: number[] }[],
+  message: string,
+): { message: string; arguements: string[] } => {
+  let negativeOffset = 0;
+  let arguements: string[] = [];
+  let sortedArguements = helpers.sort((a: any, b: any) => a.msg[0] - b.msg[0]);
+  sortedArguements.forEach((arg: any, index: number) => {
```

Comment:
`optional`: I probably prefer `for (const [index, arg of sortedArguements.entries()) {}`, mainly for readability reasons.